lib/icinga/icingaapplication.cpp

```diff
@@ -289,6 +292,40 @@ String IcingaApplication::GetNodeName() const
 	return ScriptGlobal::Get("NodeName");
 }
 
+/*
+ * Setting up PluginDir default value as a ScriptGlobal
+ *
+ * Note: only PluginDir is used inside the C++ code directly
+ * All other globals are set for compatibility, and a reduced constants.conf
+ */
+void IcingaApplication::DeclarePluginDir(const String& pluginDir)
+{
+	for (String constant : {"PluginDir", "ManubulonPluginDir", "PluginContribDir"})
+		if (!ScriptGlobal::Exists(constant))
+			ScriptGlobal::Set(constant, pluginDir);
+}
+
+String IcingaApplication::GetPluginDir(void)
+{
+	return ScriptGlobal::Get("PluginDir");
+}
+
+void IcingaApplication::DeclarePluginPath(const String& pluginPath)
+{
+	if (!ScriptGlobal::Exists("PluginPath"))
+		ScriptGlobal::Set("PluginPath", Array::FromVector(String(pluginPath).Split(":")));
+}
+
+Value IcingaApplication::GetPluginPath(void)
+{
+	Array::Ptr path = ScriptGlobal::Get("PluginPath");
+	String dir = GetPluginDir();
+	if (!dir.IsEmpty() && !path->Contains(dir))
```

Comment:
You should check that `path` isn't a nullptr before using it:

```
(lldb) bt
* thread #2, stop reason = signal SIGABRT
  * frame #0: 0x00007fff62fe5e3e libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007fff63124150 libsystem_pthread.dylib`pthread_kill + 333
    frame #2: 0x00007fff62f42312 libsystem_c.dylib`abort + 127
    frame #3: 0x00007fff62f0a368 libsystem_c.dylib`__assert_rtn + 320
    frame #4: 0x000000010000b959 icinga2`boost::intrusive_ptr<icinga::Array>::operator->(this=0x0000700008f9a7f0) const at intrusive_ptr.hpp:199
    frame #5: 0x00000001008767ab icinga2`icinga::IcingaApplication::GetPluginPath() at icingaapplication.cpp:323
    frame #6: 0x00000001008a59d4 icinga2`icinga::PluginUtility::LookupPlugin(executable=0x0000700008f9b0e8) at pluginutility.cpp:242
    frame #7: 0x00000001008a4613 icinga2`icinga::PluginUtility::ExecuteCommand(commandObj=0x0000700008f9b758, checkable=0x0000700008f9c2a8, cr=0x0000700008f9b748, macroResolvers=size=1, resolvedMacros=0x0000700008f9c288, useResolvedMacros=false, callback=0x0000000107d3ba00)> const&) at pluginutility.cpp:49
...
```

constants.conf:
```
globals.PluginPath = null
```